Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -212,7 +212,17 @@ namespace AzToolsFramework::Prefab
 
     AZ::EntityId PrefabFocusHandler::GetFocusedPrefabContainerEntityId([[maybe_unused]] AzFramework::EntityContextId entityContextId) const
     {
-        return m_focusedInstanceContainerEntityId;
+        if (m_focusedInstanceContainerEntityId.IsValid())
+        {
+            return m_focusedInstanceContainerEntityId;
+        }
+
+        if (auto instance = GetReferenceFromContainerEntityId(m_focusedInstanceContainerEntityId); instance.has_value())
+        {
+            return instance->get().GetContainerEntityId();
+        }
```

Comment:
You can directly fetch the root prefab instance here and get its container entity id here since that's the only case when m_focusedInstanceContainerEntityId will be invalid right?

Reply:
I guess this adds less number of lines and achieves the same thing...

Reply:
Yes, I mostly reused GetReferenceFromContainerEntityId to streamline the code. It gets the root prefab underneath.